frontend-vue/src/stores/subDomains.ts

```diff
@@ -0,0 +1,72 @@
+/* export a nested subdomains store from the root domains store */
+import { defineStore } from "pinia";
+import { computed, ref } from "vue";
+import { Domain, DomainId } from "~/types/domain";
+import { useDomainsStore } from "~/stores/domains";
+
+export const useSubdomainsStore = defineStore("subdomains", () => {
```

Comment:
Do we need a seperate store for subdomains or can we reuse the existing one? The naming is confusing because we don't only handle subdomains but also domains (parent).

The implementation seems to have a bug. If I do the following:
All Domains List -> Domain A -> Navigate Back (via Breadcrumbs or back button) -> Domain A = I cannot create a new subdomain.

**Two possible solutions come to mind:**
**Lookup**
Depending on how expensive it is to calculate a level for a domain it may make sens to create a initial lookup map similar to https://vscode.dev/github/trustbit/Contexture/blob/feature/beschr%C3%A4nkung-der-verschachtelungstiefe-von-subdomains/frontend-vue/src/stores/domains.ts#L112. Because we load all domains anyway for now.

So you would have a <DomainId, Level> mapping and only need to lookup based on this and don't need to recalculate it everytime you open a page (And don't need the init handling). You could also reuse the existing computed values like `domainByDomainId` and `subdomainsByDomainId` to quickly access a domain by its id instead of iterating over it.

**Disregard the store**
Reuse the existing domain store and do the domain level calculation on a component/service level. You would recalculate the level everytime you navigate to the page which may make sense if we go away from loading all domains at once.

Reply:
Good point and good catch on the side effect of the first implementation, 
I've removed the `subDomains` store and changes the code accordingly to utilize 

> <DomainId, Level> mapping 

The side-effect is fixed and all code is in `domains.ts`store